nifi/client.go

```diff
@@ -20,14 +21,34 @@ type Client struct {
 	// This breaks Terraform model to some extent but at the same time is unavoidable in NiFi world.
 	// Currently only flows that involve cross-resource interactions are wrapped into lock/unlock sections.
 	// Most of operations can still be performed in parallel.
-	Lock sync.Mutex
+	HttpScheme string
```

Comment:
I would recommend flipping those as the comment above refers to `Lock` rather than to `HttpScheme`.